libraries/ESP8266WiFi/src/include/ClientContext.h

```diff
@@ -325,6 +328,26 @@ class ClientContext
         return _write_from_source(new BufferedStreamDataSource<ProgmemStream>(stream, size));
     }
 
+    void keepAlive (uint16_t idle_sec = TCP_DEFAULT_KEEPALIVE_IDLE_SEC, uint16_t intv_sec = TCP_DEFAULT_KEEPALIVE_INTERVAL_SEC, uint8_t count = TCP_DEFAULT_KEEPALIVE_COUNT)
+    {
+        if (idle_sec && intv_sec && count) {
```

Comment:
What happens if any 2 of the 3 values are non-zero? Are all combinations allowed?

Reply:
(mostly thinking of sanity checks, if necessary)

Reply:
any 0 is an invalid value for keep-alive api, so any 0 disables keep-alive.